lib/rspec/rails/matchers/active_job.rb

```diff
@@ -148,19 +148,17 @@ def arguments_match?(job)
             end
           end
 
-          def other_attributes_match?(job)
-            serialized_attributes.all? { |key, value| value == job[key] }
-          end
+          def queue_match?(job)
+            return true unless @queue
 
-          def serialized_attributes
-            {}.tap do |attributes|
-              attributes[:at]    = serialized_at if @at
-              attributes[:queue] = @queue if @queue
-            end
+            @queue == job[:queue]
           end
 
-          def serialized_at
-            @at == :no_wait ? nil : @at.to_f
+          def at_match?(job)
+            return true unless @at
+            return job[:at].nil? if @at == :no_wait
+
+            values_match?(@at, Time.at(job[:at]))
```

Comment:
I imagine this will blow up with `TypeError: can't convert nil into an exact number` if we make an expectation of a job being scheduled in the future, but it's scheduled to be executed immediately.

Maybe a `job[:at] && values_match?(@at, Time.at(job[:at]))`?

Reply:
Write a spec and find out? If it does add a `return false if job[:at].nil? && @at != nil`?

Reply:
I'm assuming we're relying on `values_match?` here to expand the argument matcher?

Reply:
Now it's back to the same behaviour as previously

Reply:
> I'm assuming we're relying on values_match? here to expand the argument matcher?
Yep.

Reply:
> Now it's back to the same behaviour as previously

No it's just defending against the `nil` condition...

Reply:
What I mean by "back to the same behaviour" is that there's the same `expected to enqueue exactly 1 jobs, at .+ but enqueued 0` failure message in case the job is scheduled with no wait, and there's an expectation made with `at` qualifier.
Without this guard statement:
```ruby
return false unless job[:at]
```
it was erroring out with `TypeError` when trying to `Time.at` `nil` value that was in `job[:at]`.

However, I don't see that anything else needs to be done here, since `@at` at this moment is guaranteed to be neither `nil` (checked in the first guard statement) nor `:no_wait`, so it's either a date/time, either an argument matcher.

Do you have some case in mind that would misbehave?

Reply:
Your guard condition is simplier :👍: